internal/bminventory/inventory.go

```diff
@@ -2745,19 +2756,21 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 		log.WithError(err).Errorf("failed to register host <%s> cluster <%s>",
 			params.NewHostParams.HostID.String(), params.ClusterID.String())
 		uerr := errors.Wrap(err, fmt.Sprintf("Failed to register host %s", hostutil.GetHostnameForMsg(host)))
-		b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityError,
-			uerr.Error(), time.Now())
+		ev := eventgen.NewHostRegistrationFailedEvent(params.ClusterID, *params.NewHostParams.HostID, uerr.Error())
+		b.eventsHandler.SendHostEvent(ctx, ev, time.Now())
 		return returnRegisterHostTransitionError(http.StatusBadRequest, err)
 	}
 
 	if err = b.customizeHost(host); err != nil {
-		b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityError,
-			"Failed to register host: error setting host properties", time.Now())
+		ev := eventgen.NewHostRegistrationFailedEvent(params.ClusterID, *params.NewHostParams.HostID,
+			"Failed to register host: error setting host properties")
+		b.eventsHandler.SendHostEvent(ctx, ev, time.Now())
 		return common.GenerateErrorResponder(err)
 	}
 
-	b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityInfo,
-		fmt.Sprintf("Host %s: registered to cluster", hostutil.GetHostnameForMsg(host)), time.Now())
+	ev := eventgen.NewHostRegisteredClusterEvent(params.ClusterID, *params.NewHostParams.HostID,
+		fmt.Sprintf("Host %s: registered to cluster", hostutil.GetHostnameForMsg(host)))
```

Comment:
no need to include the message itself, only provide the `hostutil.GetHostnameForMsg(host)` as value for the host_name

Reply:
solved